### PR TITLE
[Misc] Log spec decode metrics

### PR DESCRIPTION
**Pull request by comaniac**
This PR adds speculative decoding metrics to engine metrics so that they can be dumped and evaluated. Two unit tests were added accordingly:
1. In `test_metrics.py`, added a unit test to verify theres metrics are correctly logged.
2. In spec decode conftest, override log interval to 0 (log per request) to log all metrics, and check the acceptance rate to be 100% when target and draft morel are the same.

cc @cadedaniel @alexm-neuralmagic

### CODE REVIEW

File: tests/spec_decode/e2e/conftest.py
Comment (cadedaniel) [inline tests/spec_decode/e2e/conftest.py:260]: nit: `ensure_all_accepted`

File: tests/spec_decode/e2e/test_multistep_correctness.py
Comment (cadedaniel) [inline tests/spec_decode/e2e/test_multistep_correctness.py:201]: nit: add docstring for new test assertion

**Review by cadedaniel — APPROVED** [review]
thanks so much for adding this

File: tests/spec_decode/e2e/conftest.py
Comment (comaniac) [reply tests/spec_decode/e2e/conftest.py:260]: Oops. Good catch.

**Comment by comaniac** [comment]
CI blocked by #6453. Will rebase once it merged.

**Pull request merged**
